src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/generalletter/GeneralLetterService.java

```diff
@@ -82,83 +71,40 @@ public CallbackResponse prepopulateData(String authorisation) {
             .build();
     }
 
-    public CallbackResponse createAndPreview(
-        CaseDetails caseDetails,
-        String authorisation,
-        String letterType,
-        String templateId
-    ) {
-        try {
-            logger.info("General Letter: creating general letter");
-            if (caseDetails.getData().containsKey(DRAFT_LETTER_DOC)) {
-                caseDetails.getData().remove(DRAFT_LETTER_DOC);
-            }
-            CCDCase ccdCase = caseDetailsConverter.extractCCDCase(caseDetails);
-            DocAssemblyResponse docAssemblyResponse = docAssemblyService.createGeneralLetter(ccdCase,
-                authorisation, templateId);
-            return AboutToStartOrSubmitCallbackResponse
-                .builder()
-                .data(ImmutableMap.of(
-                    letterType,
-                    CCDDocument.builder().documentUrl(docAssemblyResponse.getRenditionOutputLocation()).build()
-                ))
-                .build();
-        } catch (DocumentGenerationFailedException e) {
-            logger.info("General Letter creating and preview failed", e);
-            return AboutToStartOrSubmitCallbackResponse
-                .builder()
-                .errors(Collections.singletonList(ERROR_MESSAGE))
-                .build();
-        }
+    public String generateLetter(CCDCase ccdCase, String authorisation, String templateId)
+        throws DocumentGenerationFailedException {
+        logger.info("General Letter: creating general letter");
+        var docAssemblyResponse = docAssemblyService.createGeneralLetter(ccdCase, authorisation, templateId);
+        return docAssemblyResponse.getRenditionOutputLocation();
     }
 
-    public CallbackResponse printAndUpdateCaseDocuments(CaseDetails caseDetails, String authorisation) {
+    public CCDCase processDocuments(CCDCase ccdCase, Claim claim, String authorisation, String documentName) {
+        var draftLetterDoc = ccdCase.getDraftLetterDoc();
```

Comment:
This should probably have a named type as its not immediately obvious that `draftLetterDoc` is of type `CCDDocument`